app/controllers/users_controller.rb

```diff
@@ -19,46 +28,75 @@ def new
 
 private
 
-  def invite_user
+  def setup_user_in_cognito
     temporary_password = ('a'..'z').to_a.sample(3) + ('A'..'Z').to_a.sample(3) + ('0'..'9').to_a.sample(3) + ('!'..'/').to_a.sample(1)
-    begin
-      invite = SelfService.service(:cognito_client).admin_create_user(
-        temporary_password: temporary_password.join(''),
-        user_attributes: [
-          {
-            name: 'email',
-            value: @form.email
-          },
-          {
-            name: 'given_name',
-            value: @form.given_name
-          },
-          {
-            name: 'family_name',
-            value: @form.family_name
-          },
-          {
-            name: 'custom:roles',
-            value: @form.roles.join(",")
-          }
-        ],
-        username: @form.email,
-        user_pool_id: Rails.application.secrets.cognito_user_pool_id
+    SelfService.service(:cognito_client).admin_create_user(
+      temporary_password: temporary_password.join(''),
+      user_attributes: [
+        {
+          name: 'email',
+          value: @form.email
+        },
+        {
+          name: 'given_name',
+          value: @form.given_name
+        },
+        {
+          name: 'family_name',
+          value: @form.family_name
+        },
+        {
+          name: 'custom:roles',
+          value: @form.roles.join(",")
+        }
+      ],
+      username: @form.email,
+      user_pool_id: user_pool_id
       )
+  end
+
+  def add_user_to_team_in_cognito(new_user, team)
+    SelfService.service(:cognito_client).admin_add_user_to_group(
+      user_pool_id: user_pool_id,
+      username: new_user.username,
+      group_name: team.name
+    )
+  end
+
+  def invite_user
+    begin
+      if @form.team != "0"
+        team = Team.find(@form.team)
+        raise StandardError(t('users.invite.errors.team_missing')) unless Team.exists?(@form.team)
+      else
+        team = Team.new(id: 0, name: 'gds')
+      end
+      invite = setup_user_in_cognito
     rescue Aws::CognitoIdentityProvider::Errors::AliasExistsException,
            Aws::CognitoIdentityProvider::Errors::UsernameExistsException => e
       flash.now[:errors] = t('users.invite.errors.already_exists')
     rescue StandardError => e
       flash.now[:errors] = t('users.invite.errors.generic_error')
     end
 
+    begin
+      add_user_to_team_in_cognito(invite.user, team) unless team.nil?
+    rescue StandardError => e
+      flash.now[:errors] = t('users.invite.errors.generic_error')
+    end
+
     if e
       Rails.logger.error e
       render :invite, status: :bad_request
     else
       UserInvitedEvent.create(data: { user_id: invite.user.username, roles: @form.roles.join(",") })
+      UserAddedToTeamEvent.create(data: { user_id: invite.user.username, team_id: team.id, team_name: team.name })
```

Comment:
we can add the team information to the UserInvitedEvent (instead) since the user can only be invited to a team (not without)... We can then use the UserAddedToTeamEvent for actions when we move users across teams...

Reply:
Removed this line and moved the information in to the UserInvitedEvent